core/vertx-web/src/main/java/io/thorntail/vertx/web/WebRoute.java

```diff
@@ -0,0 +1,140 @@
+package io.thorntail.vertx.web;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.inject.Stereotype;
+
+import io.thorntail.vertx.web.WebRoute.WebRoutes;
+import io.vertx.core.Handler;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Annotation used to configure a {@link Route} in a declarative way. The target can be a class which implements {@link Handler} with {@link RoutingContext} as
+ * an event type or an observer method which observes {@link RoutingContext}:
+ *
+ * <pre>
+ * &#64;WebRoute("/hello")
+ * class HelloHandler implements Handler&lt;RoutingContext&gt; {
+ *
+ *     public void handle(RoutingContext ctx) {
+ *         ctx.response().setStatusCode(200).end("Hello!");
+ *     }
+ * }
+ *
+ * &#64;ApplicationScoped
+ * class Hello {
+ *
+ *     &#64;WebRoute("/hello")
+ *     void hello(&#64;Observes RoutingContext ctx) {
+ *         ctx.response().setStatusCode(200).end("Hello!");
+ *     }
+ * }
+ * </pre>
+ *
+ * <p>
+ * The annotation is repeatable, i.e. multiple annotations may be declared on a handler class or an observer method. In this case, a handler instance or an
+ * observer method is used for multiple routes.
+ * </p>
+ *
+ * <p>
+ * An annotated class which is an inner class or does not implement {@link Handler} with {@link RoutingContext} is ignored.
+ * </p>
+ *
+ * <p>
+ * Constructed handler instances are not CDI contextual intances. In other words, they're not managed by the CDI container (similarly as Java EE components like
+ * servlets). However, dependency injection, interceptors and decorators are supported.
+ * </p>
+ *
+ * <p>
+ * This annotation is annotated with {@link Stereotype} to workaround the limitations of the <code>annotated</code> bean discovery mode.
+ * </p>
+ *
+ * <p>
+ * If both {@link #value()} and {@link #regex()} are empty strings a route that matches all requests or failures is created.
+ * </p>
+ *
+ * @author Martin Kouba
+ * @see Router
+ */
+@Repeatable(WebRoutes.class)
+@Retention(RUNTIME)
+@Target({ TYPE, METHOD })
+@Stereotype
+public @interface WebRoute {
+
+    /**
+     *
+     * @see Router#route(String)
+     */
+    String value() default "";
```

Comment:
In case other parameters are often present (not sure, but looks like that's the case), would it be better to have a more descriptive name instead of the generic `value`?

Reply:
Makes sense. `@WebRoute("/path")` would probably never be used in real apps.